Fix #176 - clarify ping message responsibilities

markt-asf opened the pull request:
I plan to leave this PR open for a week or two for comments and then, depending on the comments, amend or merge the PR.

joakime commented:
I'm generally OK with this.
But do we want to talk about implementation specific pong auto-reply? (many implementations do this automatically)
Auto-pong reply is handy when you have an in-progress message and a ping arrives in the middle of it.
I believe other parts of this jakarta websocket spec doesn't allow us to call the endpoint for that middle of the message ping until the message is done.

markt-asf commented:
My understanding of the WebSocket spec is that it requires implementations to auto-reply to pings. My intention with the updated version of the second and third sentences of the ping/pong section was to make that clear. Any suggestions to further clarify that language gratefully received.

Ping and pong messages use control frames so they may be sent while other messages are in progress.

markt-asf commented:
Given that I believe the PR addresses the points raised by @joakime, I am going to go ahead and merge this PR. We can open a new issue/PR if we later determine further clarification is required.
